File: SDKs/clickmeter.com/v2/go/pkg/models/operations/datapointsget.go
package operations

import (
"openapi/pkg/models/shared")


type DataPointsGetSortDirectionEnum string

const (
    DataPointsGetSortDirectionEnumAsc DataPointsGetSortDirectionEnum = "asc"
DataPointsGetSortDirectionEnumDesc DataPointsGetSortDirectionEnum = "desc"
)



type DataPointsGetStatusEnum string

const (
    DataPointsGetStatusEnumDeleted DataPointsGetStatusEnum = "deleted"
DataPointsGetStatusEnumActive DataPointsGetStatusEnum = "active"
DataPointsGetStatusEnumPaused DataPointsGetStatusEnum = "paused"
DataPointsGetStatusEnumSpam DataPointsGetStatusEnum = "spam"
)



type DataPointsGetTypeEnum string

const (
    DataPointsGetTypeEnumTp DataPointsGetTypeEnum = "tp"
DataPointsGetTypeEnumTl DataPointsGetTypeEnum = "tl"
)


type DataPointsGetQueryParams struct {
    CreatedAfter *string `queryParam:"style=form,explode=true,name=createdAfter"`
    CreatedBefore *string `queryParam:"style=form,explode=true,name=createdBefore"`
    Limit *int32 `queryParam:"style=form,explode=true,name=limit"`
    Offset *int32 `queryParam:"style=form,explode=true,name=offset"`
    OnlyFavorites *bool `queryParam:"style=form,explode=true,name=onlyFavorites"`
    SortBy *string `queryParam:"style=form,explode=true,name=sortBy"`
    SortDirection *DataPointsGetSortDirectionEnum `queryParam:"style=form,explode=true,name=sortDirection"`
    Status *DataPointsGetStatusEnum `queryParam:"style=form,explode=true,name=status"`
    Tags *string `queryParam:"style=form,explode=true,name=tags"`
    TextSearch *string `queryParam:"style=form,explode=true,name=textSearch"`
    Type *DataPointsGetTypeEnum `queryParam:"style=form,explode=true,name=type"`
    
}

type DataPointsGetRequest struct {
    QueryParams DataPointsGetQueryParams 
    
}

type DataPointsGetResponse struct {
    APICoreResponsesEntitiesResponseAPICoreResponsesEntityURISystemInt64 *shared.APICoreResponsesEntitiesResponseAPICoreResponsesEntityURISystemInt64 
    ContentType string 
    StatusCode int64 
    
}

File: SDKs/clickmeter.com/v2/typescript/src/sdk/models/operations/datapointsget.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";

export enum DataPointsGetSortDirectionEnum {
    Asc = "asc"
,    Desc = "desc"
}

export enum DataPointsGetStatusEnum {
    Deleted = "deleted"
,    Active = "active"
,    Paused = "paused"
,    Spam = "spam"
}

export enum DataPointsGetTypeEnum {
    Tp = "tp"
,    Tl = "tl"
}


export class DataPointsGetQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=createdAfter" })
  createdAfter?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=createdBefore" })
  createdBefore?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @Metadata({ data: "queryParam, style=form;explode=true;name=offset" })
  offset?: number;

  @Metadata({ data: "queryParam, style=form;explode=true;name=onlyFavorites" })
  onlyFavorites?: boolean;

  @Metadata({ data: "queryParam, style=form;explode=true;name=sortBy" })
  sortBy?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=sortDirection" })
  sortDirection?: DataPointsGetSortDirectionEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=status" })
  status?: DataPointsGetStatusEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=tags" })
  tags?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=textSearch" })
  textSearch?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=type" })
  type?: DataPointsGetTypeEnum;
}


export class DataPointsGetRequest extends SpeakeasyBase {
  @Metadata()
  queryParams: DataPointsGetQueryParams;
}


export class DataPointsGetResponse extends SpeakeasyBase {
  @Metadata()
  apiCoreResponsesEntitiesResponseApiCoreResponsesEntityUriSystemInt64?: shared.ApiCoreResponsesEntitiesResponseApiCoreResponsesEntityUriSystemInt64;

  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;
}
